### DISCUSSION TITLE
How to use partial_combination with separator?

### DISCUSSION THREAD

**Discussion by Korsar13** [foonathan/lexy#153] [Q&A] [answered]
https://lexy.foonathan.net/playground/?id=MrT6e41E1&mode=tree

How to change grammar to parse
`id = 1, range = 0 : 100`
?

**Comment by foonathan** [comment, marked as answer]
That's not supported, and I don't know whether I want to - I've only added the combination rules for parity with other parser combinators, but am not a big fan of it.

What you can parse is a `dsl::list(identifier + dsl::lit_c<'='> + dsl::p<value>, dsl::sep(dsl::comma))`. The check that the value matches the given identifier, or that all identifiers parsed are valid is a semantic check that can be done in a later step, not a syntactic rule.

lexy is designed for parsing and that it can do a bit of semantic analysis is a coincidence and not what I consider to be best practices.
